src/ring_theory/localization.lean

```diff
@@ -436,10 +436,17 @@ ring_hom.ext $ monoid_hom.ext_iff.1 $ (to_localization_map M S).lift_of_comp j.t
 
 variables (M)
 
+/-- To show `j` and `k` agree on the whole localization, it suffices to show they agree
+on the image of the base ring, if they preserve `1` and `*`. -/
+protected lemma ext (j k : S → P) (hj1 : j 1 = 1) (hk1 : k 1 = 1)
+  (hjm : ∀ a b, j (a * b) = j a * j b) (hkm : ∀ a b, k (a * b) = k a * k b)
+  (h : ∀ a, j (algebra_map R S a) = k (algebra_map R S a)) : j = k :=
```

Comment:
With my suggestion this would become
```suggestion
  (h : j.comp (algebra_map R S).to_monoid_hom = k.comp (algebra_map R S).to_monoid_hom) : j = k :=
```